### PR TITLE
UN Trace Tool: Refactor TabRow

### PR DESCRIPTION
**Pull request by eri9000**
Related to issue: Kotlin #4773

### Summary of changes:
- Remove usages of TabRow within composables and factor it out
- Display it only when there are results available  

### Pre-merge Checklist

- a [vTest](https://runtime-kotlin.esri.com/view/all/job/vtest/job/toolkit/) Job for this PR has been run
  - [x] https://runtime-kotlin.esri.com/view/vTest/job/vtest/job/toolkit/161/
- Unit and/or integration tests have been added to exercise this PR's logic, and the tests are passing:
  - [ ] Yes
  - [ ] No

### CODE REVIEW

**Review by puneet-pdx — COMMENTED** [review]
Tabrow

@eri9000 TabRow is showing in the addStartingPointScreen. Can you please fix this.

**Comment by eri9000** [comment]
@puneet-pdx This is now fixed

File: toolkit/utilitynetworks/src/main/java/com/arcgismaps/toolkit/utilitynetworks/TraceState.kt [hunk 1]
```diff
@@ -71,6 +71,7 @@ public class TraceState(
     private val graphicsOverlay: GraphicsOverlay,
     private val mapViewProxy: MapViewProxy
 ) {
+    private val fullScreenNavRoutes = listOf(TraceNavRoute.AddStartingPoint)
```
Comment (puneet-pdx) [inline toolkit/utilitynetworks/src/main/java/com/arcgismaps/toolkit/utilitynetworks/TraceState.kt:74]: This name is confusing. Suggestion `NavRouteWithNoTabRow`

**Review by puneet-pdx — COMMENTED** [review]
@eri9000 some comments. With these changes I am seeing an undesirable flicker when deleting the trace results. 
FeatureBranch
https://github.com/user-attachments/assets/5ba9cab2-b24b-45b4-91c9-2856a81ac4b2

Delete flicker
https://github.com/user-attachments/assets/58fdd36d-bef5-4c19-b53f-3c7a2288f77d

File: toolkit/utilitynetworks/src/main/java/com/arcgismaps/toolkit/utilitynetworks/ui/TraceNavigation.kt [hunk 2]
```diff
@@ -104,15 +103,18 @@ internal fun TraceNavHost(traceState: TraceState) {
...
+                },
                 onFeatureGroupSelected = {
                     traceState.setAssetGroupName(it)
                     traceState.showScreen(TraceNavRoute.FeatureResultsDetails)
                 },
                 onDeleteResult = {
                     if (traceState.completedTraces.size == 1) {
-                        traceState.clearSelectedTraceResult()
                         traceState.showScreen(TraceNavRoute.TraceOptions)
+                        traceState.clearSelectedTraceResult()
```
[the remark this answers is not in the document]
Comment (eri9000) [reply toolkit/utilitynetworks/src/main/java/com/arcgismaps/toolkit/utilitynetworks/ui/TraceNavigation.kt]: reverted

**Review by puneet-pdx — APPROVED** [review]

File: toolkit/utilitynetworks/src/main/java/com/arcgismaps/toolkit/utilitynetworks/Trace.kt [hunk 3]
```diff
@@ -103,7 +108,21 @@ public fun Trace(
...
-                    TraceNavHost(traceState)
+                    if (traceState.showTabRow()) {
+                        TabRow(
+                            selectedTabIndex,
+                            onNavigateTo = {
+                                selectedTabIndex = it.first
+                                traceState.showScreen(it.second)
+                            }
+                        )
+                    }
```
Comment (sorenoid) [inline toolkit/utilitynetworks/src/main/java/com/arcgismaps/toolkit/utilitynetworks/Trace.kt:119]: This seems like it belongs in the `TraceNavHost` composable. What is the reason for having it here?

**Review by sorenoid — APPROVED** [review]
Looks ok. I have a question but we will revisit this

**Pull request merged**